src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx

```diff
@@ -0,0 +1,167 @@
+}
+
+const UpdateProfileForm: React.FC<UpdateProfileFormProps> = ({
+  session,
+  userFromDB,
+}) => {
+  const router = useRouter();
+
+  //cleaning up the data to load to the form
+  const learningGoalsToPass = Object.entries(userFromDB?.learningGoals).map(
+    (goal) => {
+      return { value: goal[1].goal, completed: goal[1].done };
+    }
+  );
+
+  const knowledgeLevelsToPass = Object.entries(userFromDB?.knowledgeLevels).map(
+    (item) => {
+      return { value: item[1] };
+    }
+  );
+
+  const techStackToPass = Object.entries(userFromDB?.techStack).map((item) => {
+    return { value: item[1] };
+  });
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm({
+    resolver: zodResolver(OnboardingFormSchema),
+    defaultValues: {
+      image: userFromDB?.image,
+      name: userFromDB?.name,
+      portfolio: userFromDB?.portfolio,
+      learningGoals: learningGoalsToPass,
+      location: userFromDB?.location,
+      techStack: techStackToPass,
+      knowledgeLevels: knowledgeLevelsToPass,
+      availability: userFromDB?.scheduleAvailability.available,
+      startDate: new Date(userFromDB?.scheduleAvailability.startDate),
+      endDate: new Date(userFromDB?.scheduleAvailability.endDate),
+    },
+  });
+  const [submitButtonDisabled, setSubmitButtonDisabled] = useState(false);
+
+  const showError = (message: string) => {
+    toast.error(message, {
+      position: "top-center",
+      autoClose: 5000,
+      hideProgressBar: true,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      theme: "colored",
+      progress: 0,
+      transition: Flip,
+    });
+  };
+
+  useEffect(() => {
+    const handleSingleFieldError = (prependingString: string, error: any) => {
+      let errorMsg = "";
+      if (error) {
+        errorMsg = `${prependingString} Error: ${error.message}`;
+        showError(errorMsg);
+      }
+    };
+
+    const handleFieldArrayError = (prependingString: string, error: any) => {
+      let errorMsg = "";
+      if (!error) return;
+      if (Array.isArray(error)) {
+        for (let err of error) {
+          for (let key in err) {
+            errorMsg += `${prependingString} Error: ${err[key].message} `;
+            break;
+          }
+          if (errorMsg !== "") {
+            break;
+          }
+        }
+      } else {
+        errorMsg = `${prependingString} Error: ${error.message}`;
+      }
+      showError(errorMsg);
+    };
+    console.log(errors);
+    if (Object.keys(errors).length !== 0) {
+      handleSingleFieldError("Name", errors.name);
+      if (errors.portfolio) {
+        handleSingleFieldError("Portfolio", errors.portfolio);
+      }
+      handleFieldArrayError("Learning Goals", errors.learningGoals);
+      handleFieldArrayError("Knowledge Levels", errors.knowledgeLevels);
+    }
+  }, [errors]);
+
+  const onSubmit = async (data: any, event: any) => {
+    console.log(data);
```

Comment:
```suggestion
```
Let's remove this testing log once we're done here.

If we keep all of these logs in, by the time we're done with the project we'll have a lot of trouble debugging any issue!